crates/ubrn_cli/src/android.rs

```diff
@@ -136,9 +148,74 @@ impl AndroidArgs {
 
             cmd.args(android.cargo_extras.clone());
 
-            run_cmd(cmd.current_dir(&rust_dir))?
+            run_cmd(cmd.current_dir(&rust_dir))?;
+            let metadata = crate_.metadata()?;
+            let src_lib = metadata.library_path(
+                Some(target.triple()),
+                CrateMetadata::profile(self.common_args.release),
+            )?;
+            let dst_dir = jni_libs.join(target.to_string());
+            mk_dir(&dst_dir)?;
+
+            let dst_lib = dst_dir.join(metadata.library_file(Some(target.triple())));
+            fs::copy(&src_lib, &dst_lib)?;
         }
 
         Ok(())
     }
 }
+
+#[derive(Debug, Deserialize, Default, Clone)]
+pub enum Target {
+    #[serde(rename = "armeabi-v7a")]
+    ArmeabiV7a,
+    #[default]
+    #[serde(rename = "arm64-v8a")]
+    Arm64V8a,
+    #[serde(rename = "x86")]
+    X86,
+    #[serde(rename = "x86_64")]
+    X86_64,
+}
+
+impl FromStr for Target {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            // match android style architectures
+            "armeabi-v7a" => Target::ArmeabiV7a,
+            "arm64-v8a" => Target::Arm64V8a,
+            "x86" => Target::X86,
+            "x86_64" => Target::X86_64,
+            // match rust triple architectures
+            "armv7-linux-androideabi" => Target::ArmeabiV7a,
+            "aarch64-linux-android" => Target::Arm64V8a,
+            "i686-linux-android" => Target::X86,
+            "x86_64-linux-android" => Target::X86_64,
+            _ => return Err(anyhow::anyhow!("Unsupported target: '{s}'")),
+        })
+    }
+}
+
+impl Display for Target {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(match self {
+            Target::ArmeabiV7a => "armeabi-v7a",
+            Target::Arm64V8a => "arm64-v8a",
+            Target::X86 => "x86",
+            Target::X86_64 => "x86_64",
+        })
+    }
+}
+
+impl Target {
+    pub fn triple(&self) -> &'static str {
```

Comment:
Just for my own understanding, what is the history/implication of the name 'triple' here?

Reply:
The target triple is the`cpu-vendor-os` that we're compiling for.